File: source/frontend/src/models/api/Agreement.ts
import { Api_AuditFields } from './AuditFields';
import { Api_ConcurrentVersion } from './ConcurrentVersion';
import Api_TypeCode from './TypeCode';

export interface Api_Agreement extends Api_ConcurrentVersion, Api_AuditFields {
  agreementId: number;
  acquisitionFileId: number;
  agreementType: Api_TypeCode<string>;
  agreementStatusType: Api_TypeCode<string>;
  agreementDate: string | null;
  completionDate: string | null;
  terminationDate: string | null;
  commencementDate: string | null;
  possessionDate: string | null;
  depositAmount: number | null;
  noLaterThanDays: number | null;
  purchasePrice: number | null;
  legalSurveyPlanNum: string | null;
  offerDate: string | null;
  expiryDateTime: string | null;
  signedDate: string | null;
  inspectionDate: string | null;
  cancellationNote: string | null;
}

export enum AgreementTypes {
  H0074 = 'H0074',
  H179A = 'H179A',
  H179P = 'H179P',
  H179T = 'H179T',
}

export enum AgreementStatusTypes {
  DRAFT = 'DRAFT',
  FINAL = 'FINAL',
  CANCELLED = 'CANCELLED',
}

File: source/backend/entities/Partials/Agreement.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Pims.Dal.Entities
{
    /// <summary>
    /// PimsAgreement class, provides an entity for the datamodel to manage agreements.
    /// </summary>
    public partial class PimsAgreement : StandardIdentityBaseAppEntity<long>, IBaseAppEntity
    {
        #region Properties
        [NotMapped]
        public override long Internal_Id { get => this.AgreementId; set => this.AgreementId = value; }
        #endregion

        public bool IsEqual(PimsAgreement other)
        {
            return AgreementId == other.AgreementId &&
                AcquisitionFileId == other.AcquisitionFileId &&
                AgreementTypeCode == other.AgreementTypeCode &&
                AgreementStatusTypeCode == other.AgreementStatusTypeCode &&
                AgreementDate == other.AgreementDate &&
                CompletionDate == other.CompletionDate &&
                TerminationDate == other.TerminationDate &&
                CommencementDate == other.CommencementDate &&
                DepositAmount == other.DepositAmount &&
                NoLaterThanDays == other.NoLaterThanDays &&
                PurchasePrice == other.PurchasePrice &&
                LegalSurveyPlanNum == other.LegalSurveyPlanNum &&
                OfferDate == other.OfferDate &&
                ExpiryTs == other.ExpiryTs &&
                SignedDate == other.SignedDate &&
                InspectionDate == other.InspectionDate &&
                ExpropriationDate == other.ExpropriationDate &&
                PossessionDate == other.PossessionDate &&
                CancellationNote == other.CancellationNote;
        }
    }
}
